pyogrio/geopandas.py

```diff
@@ -168,8 +168,11 @@ def write_dataframe(df, path, layer=None, driver=None, encoding=None, **kwargs):
     # TODO: may need to fill in pd.NA, etc
     field_data = [df[f].values for f in fields]
 
-    # TODO: validate geometry types, not all combinations are valid
-    geometry_type = geometry.type.unique()[0]
+    if not df.empty:
+        # TODO: validate geometry types, not all combinations are valid
+        geometry_type = geometry.type.unique()[0]
+    else:
+        geometry_type = "Unknown"
```

Comment:
`"Unknown"` is what OGR returns in `read_info` if you try to read an empty file saved by fiona and ensures that the behaviour of both is the same. For some reason, shapefile defaults to LineString (in both fiona and pyogrio) but the rest is read back correctly as "Unknown" geometry type.